lib/Listener/CalendarListener.php

```diff
@@ -59,13 +60,16 @@ public function handle(Event $event): void {
 		$collectionName = $event->getCalendarData()['uri'];
 		$subscriptions = $this->subscriptionService->findAll($collectionName);
 
-		foreach($subscriptions as $subscription) {
-			$transport = $this->transportManager->getTransport($subscription->getTransport());
-			try {
-				$transport->notify($subscription->getUserId(), $collectionName, $subscription->getId());
-			} catch (\Exception $e) {
-				$this->logger->error("transport " .  $subscription->getTransport() . " failed to deliver notification to subscription " . $subscription->getId());
+		$notificationPromises = (function () use ($collectionName, $subscriptions): Generator {
```

Comment:
```suggestion
		$notificationPromises = (function () use ($collectionName, $subscriptions): \Generator {
```

Or use/import `Generator` above.

Error: `Generator return type must be a supertype of Generator, OCA\DavPush\Listener\Generator given at /var/www/html/custom_apps/dav_push/lib/Listener/CalendarListener.php#63`